DEVELOPERS_GUIDE.md

```diff
@@ -0,0 +1,87 @@
+
+String CONCURRENT_TIME = PAUSE_TIME;
+String PAUSE_TIME = TIME + "\\s?(?:secs?|ms)";
+String TIME = "(-?" + REAL_NUMBER + “)";
+String REAL_NUMBER = INTEGER + DECIMAL_POINT + INTEGER;
+String DECIMAL_POINT = "(?:\\.|,)";
+String INTEGER = "\\d+";
+
+Since many lines contain either a pause or concurrent time, defining these “tokens” eases the complexity of creating and debugging the parse rules.
+
+####Testing
+
+There are two main types of tests that are used to QA parsers and rules. The first is known as a rules capture test. In this test the rule is exposed to everything that it is likely to encounter. The test passes if the rule captures everything it should capture and doesn’t capture anything else. In the former test, the parser is configured to send generated events to the test. The test the counts each of the different types of events. The test passes if the counts for each event type reach the expected value.
+
+There is a third test that has, unfortunately, not bee applied to all parsers as of yet. In this test, the parsers will parse a log fragment to produce a well known event. The test passes if all of the attributes of that event match the expected value.
+
+
+For historical reasons, SurvivorSpace is treated as a JVMEvent even though in reality, it’s an attribute of a young generational collection. This record is created by a parser that is specialized to parse the log and extract out the survivor information.
+
+### JVMEvent
+
+A JVMEvent represents something that happened in the JVM. The primary attributes are the type of event, the time of the event, and it’s duration. Each specialization of JVMEvent will contain attributes that are specific to that event. For example, a PSYoung represent a Parallel Young generational collection. In addition to the information above, it will have heap occupancy before and after the collection along with the size of heap before and after the collection. The event may contain more information depending upon the level of detailed information in the log file. As previously mentioned, any survivor space data will be ignored as that is dealt with in its own event.  A complete set of events can be seen in the diagram below. A complete discription of each event can be found in the corresponding JavaDoc.
+
+![GCToolkit Design](images/GCToolkit_Events.png)
+Figure 2. GCToolkit Event Hierarchy
+
+### Aggregator and Aggregation
+
+
+As can be seen in Figure 1, the end point for delivery of a JVMEvent as far as the framework is concerned, is an Aggregator. Aggregator classes must be provided by the users of GCToolkit as they are not provided as they are considered out of scope for this project. The role of the Aggregator is to capture the JVMEvent objects that are of interest to it and the extract the data of interest and pass it along to an Aggregation.
```

Comment:
s/JVMEvent as far/JVMEvent, as far/ or reword as 'As far as the framework is concerned, the endpoint for delivery of a JVMEvent is an Aggregator, as can be seen in Figure 1.'
I would move the last sentence before the sentence about Aggregator classes being out of scope and the sentence in the next paragraph about Aggregations being supplied by the user into a separate paragraph at the end of this section. And then point the reader to the sample.